linux/gtk/src/main/cpp/DarkModeGtk.cpp

```diff
@@ -63,21 +65,24 @@ struct EventHandler {
     }
 
     void stop() {
-        app->release();
         app->quit();
-        loopThread.join();
+        app->release();
+
         settingsChangedSignalConnection.disconnect();
+        loopThread.join();
     }
 
     void run() {
-        app->hold();
         app->run();
     }
 
     EventHandler(JavaVM *jvm_, jobject callback_) {
         jvm = jvm_;
         callback = callback_;
 
+        app = Gtk::Application::create();
+        app->hold();
```

Comment:
we should also move `settings = Gtk::Settings::get_default();` to be after `app` has been created. (otherwise you get null it seems)